Content Version Cleanup (#3640)
umbraco/UmbracoDocs: Reference/V9-Config/ContentSettings/index-v9.0.0.md

---
versionFrom: 9.0.0
meta.Title: "Umbraco Content Settings"
meta.Description: "Information on the content settings section"
state: complete
verified-against: beta-3
update-links: true
---

# Content Settings

> Content settings contains various settings regarding content in the CMS, such as allowed upload files, image settings, and much more. All the values in the content settings has default values, so all configuration is optional.
> 
> To get an overview of the keys and values in the global section, the following snippet will contain all available options, along with their default values:

Reviewer: ```suggestion
Content settings contain various settings regarding the content in the CMS, such as allowed upload files, image settings, and much more. All the values in the content settings have default values, so all configuration is optional.

To get an overview of the keys and values in the global section, the following snippet will contain all available options, along with their default values:
```

Author: Also unchanged by this PR

```json
"Umbraco": {
  "CMS": {
    "Content": {
      "ResolveUrlsFromTextString": false,
      "Error404Collection": [],
      "PreviewBadge": "",
      "MacroErrors": "Inline",
      "DisallowedUploadFiles": ["ashx", "aspx", "ascx", "config", "cshtml", "vbhtml", "asmx", "air", "axd", "xamlx"],
      "AllowedUploadFiles": [],
      "ShowDeprecatedPropertyEditors": false,
      "LoginBackgroundImage": "~/assets/img/login.jpg",
      "LoginLogoImage": "~/assets/img/application/umbraco_logo_white.svg",
      "Notifications": {
        "Email": "",
        "DisableHtmlEmail": false
      },
      "Imaging": {
        "ImageFileTypes": ["jpeg", "jpg", "gif", "bmp", "png", "tiff", "tif"],
        "AutoFillImageProperties": {
          "Alias": "umbracoFile",
          "WidthFieldAlias": "umbracoWidth",
          "HeightFieldAlias": "umbracoHeight",
          "LengthFieldAlias": "umbracoBytes",
          "ExtensionFieldAlias": "umbracoExtension"
        }
      }
    }
  }
}
```

> ## Root level settings
> 
> In the root level section, that is those without a seperate sub section like Imaging, you can configure:
> 
> ### Resolve urls from text string
> 
> This setting is used when you're running Umbraco in virtual directories. Setting this to true can increase render time for pages with a large number of links, however, this is required if Umbraco is running in a virtual directory.
> 
> ### Error 404 collection
> 
> In case of a 404 error (page not found) Umbraco can return a default page instead. This is set here. Notice you can also set a different error page, based on the current culture so a 404 page can be returned in the correct language.
> 
> ```json
> "Error404Collection": [{
>   "ContentId": 1,
>   "Culture": "en-US"
> }]
> ```
> 
> The above example shows what you need to do if you only have a single site that needs to show a custom 404 page. You specify which node that should be shown when a request for a non-existing page is being made. You can specify the node in three ways:

Reviewer: ```suggestion
## Root-level settings

In the root level section, that is those without a separate subsection like Imaging, you can configure:

### Resolve URLs from a text string

This setting is used when you're running Umbraco in virtual directories. Setting this to true can increase render time for pages with a large number of links, however, this is required if Umbraco is running in a virtual directory.

### Error 404 collection

In case of a 404 error (page not found) Umbraco can return a default page instead. This is set here. Notice you can also set a different error page, based on the current culture so a 404 page can be returned in the correct language.

```json
"Error404Collection": [{
  "ContentId": 1,
  "Culture": "en-US"
}]
```

The above example shows what you need to do if you only have a single site that needs to show a custom 404 page. You specify which node that should be shown when a request for a non-existing page is being made. You can specify the node in three ways:
```

Author: This is unchanged by my PR :p
